microprofile/fault-tolerance/src/main/java/io/helidon/microprofile/faulttolerance/MethodIntrospector.java

```diff
@@ -194,26 +203,47 @@ Bulkhead getBulkhead() {
     }
 
     /**
-     * Search for annotation first on the method and then its class. This method
-     * does not check if the annotation's target is of element type {@link
-     * java.lang.annotation.ElementType#TYPE}.
-     *
-     * @param clazz Annotation class to search for.
-     * @param <T> Annotation type.
-     * @return Annotation instance or {@code null} if not found.
-     */
-    private <T extends Annotation> T getAnnotation(Class<T> clazz) {
-        final T annotation = method.getAnnotation(clazz);
-        return annotation != null ? annotation : method.getDeclaringClass().getAnnotation(clazz);
-    }
-
-    /**
-     * Determines if annotation type is present on the method or its class.
+     * Determines if annotation type is present and enabled.
      *
      * @param clazz Annotation class to search for.
      * @return Outcome of test.
      */
-    private boolean isAnnotationPresent(Class<? extends Annotation> clazz) {
-        return getAnnotation(clazz) != null;
+    private boolean isAnnotationEnabled(Class<? extends Annotation> clazz) {
+        LookupResult<? extends Annotation> lookupResult = lookupAnnotation(beanClass, method, clazz);
+        if (lookupResult == null) {
+            return false;       // not present
+        }
+
+        String value;
+        final String annotationType = clazz.getSimpleName();
+
+        // Check if property defined at method level
+        String methodLevel = String.format("%s/%s/%s/enabled",
```

Comment:
At the risk of being accused of advocating over-engineering... The property name patterns (for method- and class-level and the global level) are used both here and in `MethodAntn.lookupAnnotation` (and maybe in later code in the review I haven't read yet). 

Does it make sense to create one or more utility methods somewhere that would accept the property name and the (optional?) class and method names, encapsulate the details of how to construct the full prop. name from them, and return the fully-qualified property name?

Reply:
Yes, I think some code could be factored out here, that would at least not repeat the call to `String.format()`. If you agree, let's merge it as is and I will provide an enhancement for this. Unfortunately, the build takes over 30 mins now with 214 TCK FT tests :(

Reply:
Ouch! Yes, I say "ship it" as-is for now.